hazelcast/src/main/java/com/hazelcast/config/RingbufferConfig.java

```diff
@@ -0,0 +1,322 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import com.hazelcast.spi.annotation.Beta;
+
+import static com.hazelcast.config.InMemoryFormat.NATIVE;
+import static com.hazelcast.util.Preconditions.checkAsyncBackupCount;
+import static com.hazelcast.util.Preconditions.checkBackupCount;
+import static com.hazelcast.util.Preconditions.checkFalse;
+import static com.hazelcast.util.Preconditions.checkNotNegative;
+import static com.hazelcast.util.Preconditions.checkNotNull;
+import static com.hazelcast.util.Preconditions.checkPositive;
+
+/**
+ * Contains the configuration for the {@link com.hazelcast.ringbuffer.Ringbuffer}.
+ * <p/>
+ * The RingBuffer is currently not a distributed data-structure, so its content will be fully stored on a single member
+ * in the cluster and its backup in another member in the cluster.
+ */
+@Beta
+public class RingbufferConfig {
+
+    /**
+     * Default value of capacity of the RingBuffer.
+     */
+    public static final int DEFAULT_CAPACITY = 100 * 1000;
+    /**
+     * Default value of synchronous backup count
+     */
+    public static final int DEFAULT_SYNC_BACKUP_COUNT = 1;
+    /**
+     * Default value of asynchronous backup count
+     */
+    public static final int DEFAULT_ASYNC_BACKUP_COUNT = 0;
+    /**
+     * Default value for the retention period.
+     */
+    public static final int DEFAULT_RETENTION_SECONDS = 30;
```

Comment:
The problem with no retention is that the data is not cleared. Only due to overwrite. So it behaves like an array and this can lead to people complaining about memory usage since memory is retained.